python/tvm/topi/nn/depthwise_conv2d.py

```diff
@@ -252,8 +253,18 @@ def depthwise_conv2d_nhwc(Input, Filter, stride, padding, dilation, out_dtype=No
         dilation_h, dilation_w = dilation
 
     batch, in_height, in_width, in_channel = Input.shape
+
+    dim = len(Input.shape) - 2
+
     # shape of dilated kernel
-    filter_height, filter_width, filter_channel, channel_multiplier = Filter.shape
+    if kernel_layout == "HWOI":
+        filter_height, filter_width, filter_channel, channel_multiplier = Filter.shape
+        kernel_permutation_to = [0, 1] + list(range(2, dim + 2))
+    elif kernel_layout == "HWIO":
+        filter_height, filter_width, channel_multiplier, filter_channel = Filter.shape
+        kernel_permutation_to = [dim + 1, dim] + list(range(dim))
```

Comment:
I'm not seeing how this permutation generates `HWIO`.  This defines `kernel_permutation_to` as `[3, 2, 0, 1]`, so `kernel_permutation_from` is `[2, 3, 1, 0]`.  With the usage below, that would permute from `[di, dj, c//channel_multiplier, c%channel_multiplier]` to `[c//channel_multiplier, c%channel_multiplier, dj, di]`, which would be `OIWH`.

Should this be `list(range(dim)) + [dim + 1, dim]` instead?

Reply:
@Lunderberg  You are right, my bad. Not sure why I thought this would result in `HWIO`. Thanks so much for catching this bug.